diff --git a/progs/mkcapshdoc.sh b/progs/mkcapshdoc.sh
index d2ee4bd..3f14d40 100755
--- a/progs/mkcapshdoc.sh
+++ b/progs/mkcapshdoc.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh -e
 # This script generates some C code for inclusion in the capsh binary.
 # The Makefile generally only generates the .c code and compares it
 # with the checked in code in the progs directory.
@@ -13,12 +13,16 @@ cat<<EOF
  */
 EOF

-let x=0
-while [ -f "../doc/values/${x}.txt" ]; do
+num_files=-1
+
+for _ in ../doc/values/*.txt; do
+    : $((num_files+=1))
+done
+
+for x in $(seq 0 "${num_files}"); do
     name=$(grep -F ",${x}}" ../libcap/cap_names.list.h|sed -e 's/{"//' -e 's/",/ = /' -e 's/},//')
     echo "static const char *explanation${x}[] = {  /* ${name} */"
     sed -e 's/"/\\"/g' -e 's/^/    "/' -e 's/$/",/' "../doc/values/${x}.txt"
-    let x=1+${x}
     echo "    NULL"
     echo "};"
 done
@@ -26,13 +30,12 @@ done
 cat<<EOF
 const char **explanations[] = {
 EOF
-let y=0
-while [ "${y}" -lt "${x}" ]; do
+
+for y in $(seq 0 "${num_files}"); do
     echo "    explanation${y},"
-    let y=1+${y}
 done
 cat<<EOF
 };

-const int capsh_doc_limit = ${x};
+const int capsh_doc_limit = $((num_files+1));
 EOF
